meson.build

```diff
@@ -3,13 +3,38 @@ project('abseil-cpp', 'cpp',
   license : 'Apache-2.0',
   default_options : [
     'cpp_std=c++11',
+    'warning_level=0'
   ])
 
 cpp = meson.get_compiler('cpp')
 
 cpp_flags = [
-  '-Wno-sign-compare'
+  '-Wcast-qual',
+  '-Wconversion-null',
+  '-Wformat',
+  '-Wformat-security',
+  '-Wmissing-declarations',
+  '-Woverlength-strings',
+  '-Wpointer-arith',
+  '-Wundef',
+  '-Wunused-local-typedefs',
+  '-Wunused-result',
+  '-Wvarargs',
+  '-Wvla',
+  '-Wwrite-strings',
+  '-DNOMINMAX',
```

Comment:
Where are these compile flag coming from? I got lost in the Abseil CMake maze but I was not able to find them. Maybe it could be worth to add a comment stating where to look if these need to be updated in the future.

Reply:
Sure I can add a comment. Check https://github.com/abseil/abseil-cpp/tree/master/absl/copts

They generate Cmake and Bazel copts at the same time (but not meson :( )

Reply:
I don't understand how you condense the options defined there into the options in this list. These seem to correspond to `ABSL_GCC_FLAGS` in the CMake file but I don't understand why you decided to take the GCC options and not the Clang or the MSVC options. As these are only compiler warning options I would simply leave them out for all compilers.

Reply:
Didn't give it much though to be honest. I agree better leave them empty.